src/compiler/checker.ts

```diff
@@ -24440,9 +24498,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return undefined;
     }
 
-    function isTupleTypeStructureMatching(t1: TupleTypeReference, t2: TupleTypeReference) {
+    function isTupleTypeStructureMatching(t1: TupleTypeReference, t2: TupleTypeReference, tupleStructureComparisonKind: TupleStructureComparisonKind) {

```

Comment:
Curious if we could just accept `ElementFlags` as a mask here instead of coming up with a new enum.